### PR TITLE
Migrate DB to use origin and formActionOrigin

### PR DESCRIPTION
**Pull request by skhamis**
Since we're going to be migrating the DB anyways we might as well get rid of any mention of hostname/formSubmitUrl since desktop is now using origin/formActionOrigin and that is the "proper" name for these values anyways. I didn't need to actually/procure are migration code since when we create the tables it'll have the new names anyways.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/logins/src/schema.rs [hunk 1]
```diff
@@ -325,8 +325,8 @@ pub fn upgrade_sqlcipher_db(db: &mut Connection, encryption_key: &str) -> Result
     if user_version < 3 {
         // These indices were added in v3 (apparently)
         tx.execute_all(&[
-            CREATE_OVERRIDE_HOSTNAME_INDEX_SQL,
-            CREATE_DELETED_HOSTNAME_INDEX_SQL,
+            CREATE_OVERRIDE_ORIGIN_INDEX_SQL,
+            CREATE_DELETED_ORIGIN_INDEX_SQL,
         ])?;
```
Comment (bendk) [inline components/logins/src/schema.rs:330]: I think these should stay the same.  If a user actually had a v2 database, then they would be using the old column names at this point.

**Review by bendk — CHANGES_REQUESTED** [review]
The new code looks good overall and should work for new DBs, but I do think you need code to upgrade old databases.

The test I used a bunch was running the `sync-pass` example  with a checkout from `main`, adding a login or two, then checking out this branch and running `sync-pass` with the same database.  That should trigger the schema upgrade code and I think you'll see issues.

I think you could get away without writing a full new upgrade though.  The v4 -> v5 upgrade hasn't been released yet, so you could maybe sneak in the extra rename column statements around line 397.

File: components/logins/src/schema.rs (hunk 1 above)
Comment (skhamis) [reply components/logins/src/schema.rs]: Good catch, was a bit over zealous in the renames. Turns out I need to have a new version of that function for the `create` while keeping that one for any v3. Tests pass and migrations work now!

**Comment by mhammond** [comment]
I can't see the PR for logins-per-field-encryption - where does that live? I'm not yet convinced `sqlcipher_export()` is the correct approach here given the other schema and encryption changes, but I'm not sure where to add those comments :)

**Comment by bendk** [comment]
> I can't see the PR for logins-per-field-encryption - where does that live? I'm not yet convinced `sqlcipher_export()` is the correct approach here given the other schema and encryption changes, but I'm not sure where to add those comments :)

That one was https://github.com/mozilla/application-services/pull/4135, but I closed it when I merged the code from my fork into the mozilla branch.  It would be great to talk about that approach more, maybe we could just re-open the PR and continue there.

**Comment by bendk** [comment]
Looks good to me after the latest changes.

**Pull request merged**